hypershift-operator/controllers/nodepool/nto.go

```diff
@@ -70,13 +72,41 @@ func (r *NodePoolReconciler) reconcileMirroredConfigs(ctx context.Context, logr
 	// delete the redundant configs that are no longer part of the nodepool spec
 	for i := 0; i < len(existingConfigsList.Items); i++ {
 		existingConfig := &existingConfigsList.Items[i]
-		if toDelete.Has(existingConfig.Name) {
-			_, err := supportutil.DeleteIfNeeded(ctx, r.Client, existingConfig)
-			if err != nil {
-				return fmt.Errorf("failed to delete ConfigMap %s/%s: %w", existingConfig.Namespace, existingConfig.Name, err)
-			}
+		if !toDelete.Has(existingConfig.Name) {
+			continue
+		}
+		_, err := supportutil.DeleteIfNeeded(ctx, r.Client, existingConfig)
+		if err != nil {
+			return fmt.Errorf("failed to delete ConfigMap %s/%s: %w", existingConfig.Namespace, existingConfig.Name, err)
 		}
 	}
+	// NTO also generates config in the HCP namespace.
+	ntoGeneratedKubeletConfigs := &corev1.ConfigMapList{}
+	if err := r.List(ctx, ntoGeneratedKubeletConfigs, &client.ListOptions{
+		Namespace: controlPlaneNamespace,
+		LabelSelector: labels.SelectorFromValidatedSet(labels.Set{
+			nodeTuningGeneratedConfigLabel: "true",
+			KubeletConfigConfigMapLabel:    "true",
+			hyperv1.NodePoolLabel:          nodePool.Name,
+		}),
+	}); err != nil {
+		return err
+	}
+	// we need to validate that generated configs and user-provided configs
+	// are not conflicting with each other, before we create the new ones
+	isValid, reason := validateMirroredConfigs(ntoGeneratedKubeletConfigs.Items, mirroredConfigs, nodePool.Name)
```

Comment:
why is not validateMirroredConfigs just returning an error, and constructing the error message with the list of names? so then this is also bubbled up in to the condition that catches the error when invoking this function.